pkg/users/group-details.js

```diff
@@ -0,0 +1,185 @@
+        if (undoItem.type === 'added') {
+            removeAccount(undoItem.name, true).then(() => setHistory(history.slice(0, -1)));
+        } else if (undoItem.type === 'removed') {
+            addAccount(undoItem.name, true).then(() => setHistory(history.slice(0, -1)));
+        }
+    };
+
+    const removeAccount = (account, isUndo) => {
+        if (!isUndo)
+            setHistory([...history, { type: 'removed', name: account }]);
+
+        const current_account = accounts.find(a => a.name == account);
+        if (current_account.loggedIn)
+            setLoggedInUsersAffected(true);
+
+        return cockpit.spawn(["gpasswd", "-d", account, group], { superuser: "require", err: "message" })
+                .then(() => {
+                    setSelected(selected.filter(item => item !== account));
+                    setIsOpenAccount(false);
+                }, show_unexpected_error);
+    };
+
+    const addAccount = (account, isUndo) => {
+        if (!isUndo)
+            setHistory([...history, { type: 'added', name: account }]);
+
+        const current_account = accounts.find(a => a.name == account);
+        if (current_account.loggedIn) setLoggedInUsersAffected(true);
+
+        return cockpit.spawn(["gpasswd", "-a", account, group], { superuser: "require", err: "message" })
+                .then(() => {
+                    setSelected([...selected, account]);
+                    setIsOpenAccount(false);
+                }, show_unexpected_error);
+    };
+
+    const onSelectAccount = (event, selection) => {
+        if (selected.includes(selection)) {
+            removeAccount(selection);
+        } else {
+            addAccount(selection);
+        }
+    };
+
+    const chipGroupComponent = () => {
+        return (
+            <LabelGroup numLabels={10}>
+                {(selected || []).map((currentLabel, index) => {
+                    const optional = !current_group.userlistPrimary.includes(currentLabel) && superuser.allowed ? { onClose: () => removeAccount(currentLabel) } : {};
+
+                    return (
+                        <Label key={currentLabel}
+                               color={accounts.find(ac => ac.name === currentLabel).isAdmin ? "gold" : "cyan"}
+                               {...optional}
+                        >
+                            {currentLabel}
+                        </Label>
+                    );
+                })}
+            </LabelGroup>
+        );
+    };
+
+    return (
+        <FormGroup
+            fieldId="group-accounts"
+            helperText={
+                (history.length > 0)
+                    ? <HelperText className="pf-c-form__helper-text">
+                        <Flex>
+                            {loggedInUsersAffected && <HelperTextItem id="group-accounts-helper" variant="warning">{_("The logged in users must log out and log back in for the new configuration to take effect.")}</HelperTextItem>}
```

Comment:
This is both very long (translations will become even longer), and also redundant. It has "log" three times. Maybe something like

> To take effect, these users must log out.

Reply:
the 'these users must log out' is not entirely correct, as it refers to the users shown in the labels, but if some user was removed from the group, (label was removed) then the user also needs to log out for the configuration to take effect.

Reply:
How about 'To take effect, the affected users must log out'

Reply:
WFM. Possibly "changed" users?